scripts/script.js

```diff
@@ -0,0 +1,102 @@
+// Fetch and display logic ------------------------------------>
+// Fetch--------------------------->
+const fetchURL = "http://localhost:3000/products";
+
+const fetchData = () => {
+  fetch(fetchURL)
+    .then((res) => {
+      return res.json();
+    })
+    .then((data) => localStorage.setItem("fetchedData", JSON.stringify(data)));
+};
+
+const checkIfFetched = () => {
+  if (localStorage.getItem("fetchedData") === null) {
+    fetchData();
+  }
+};
+
+document.getElementsByTagName("body").onload = checkIfFetched();
+
+// Display-------------------------->
+const getButtonId = (idVal) => {
+  let clickedId = idVal;
+  localStorage.setItem("clickedId", JSON.stringify(clickedId));
+};
+
+let clickedId = JSON.parse(localStorage.getItem("clickedId"));
+let displayData = JSON.parse(localStorage.getItem("fetchedData"));
+
+const display = () => {
+  displayData.map((el) => {
+    if (el.id == clickedId) {
```

Comment:
see if you can use strict equality here: https://github.com/airbnb/javascript#user-content-comparison-operators--equality